### PR TITLE
Add doctest and fix mypy type annotation in bellman ford

### PR DESCRIPTION
**Pull request by mhihasan**
### **Describe your change:**
- Add doctest and 
- fix mypy type annotation in bellman ford

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

related issue #4052

### CODE REVIEW

File: graphs/bellman_ford.py [hunk 1]
```diff
@@ -1,56 +1,53 @@
-from __future__ import annotations
+from typing import Dict, List
```
Comment (cclauss) [inline graphs/bellman_ford.py:1]: This is the opposite of what we want.

File: graphs/bellman_ford.py [hunk 2]
```diff
@@ -1,56 +1,53 @@
...
 
-    for i in range(E):
-        graph[i][i] = 0.0
+if __name__ == "__main__":
+    edges = [(2, 1, -10), (3, 2, 3), (0, 3, 5), (0, 1, 4)]
+    g = [{"src": s, "dst": d, "weight": w} for s, d, w in edges]
+    shortest_distance = bellman_ford(g, 4, 4, 0)
+    print_distance(shortest_distance, 0)
 
-    for i in range(E):
```
Comment (cclauss) [inline graphs/bellman_ford.py:48]: Why remove user interaction?

File: graphs/bellman_ford.py [hunk 3]
```diff
@@ -1,56 +1,53 @@
...
 
 
-def BellmanFord(graph: list[dict[str, int]], V: int, E: int, src: int) -> int:
+def bellman_ford(
+    graph: List[Dict[str, int]], vertex_count: int, edge_count: int, src: int
+) -> List[float]:
     """
     Returns shortest paths from a vertex src to all
     other vertices.
+    >>> edges = [(2, 1, -10), (3, 2, 3), (0, 3, 5), (0, 1, 4)]
```
Comment (cclauss) [inline graphs/bellman_ford.py:16]: Why not just create `test_edges` at global scope and then use them here and also at line 46.

File: graphs/bellman_ford.py [hunk 4]
```diff
@@ -1,56 +1,53 @@
...
+    distance[src] = 0.0
 
-    for i in range(V - 1):
-        for j in range(E):
-            u = graph[j]["src"]
-            v = graph[j]["dst"]
-            w = graph[j]["weight"]
+    for i in range(vertex_count - 1):
+        for j in range(edge_count):
+            u, v, w = graph[j]["src"], graph[j]["dst"], graph[j]["weight"]
```
Comment (cclauss) [inline graphs/bellman_ford.py:31]:
```suggestion
            src, dst, weight = graph[j][field] for field in ("src", "dst", "weight")
```

File: graphs/bellman_ford.py (hunk 1 above)
Comment (mhihasan) [reply graphs/bellman_ford.py:1]: Do we want to remove type annotation? I think we need to fix mypy annotation, but not to remove.

File: graphs/bellman_ford.py (hunk 2 above)
Comment (mhihasan) [reply graphs/bellman_ford.py:48]: If user mistakes any input, he/she will needs to restart again which is cumbersome. Instead, user(obviously a programmer) can provide graph providing as a list of dictionary.

File: graphs/bellman_ford.py (hunk 3 above)
Comment (mhihasan) [reply graphs/bellman_ford.py:16]: Will look into it

File: graphs/bellman_ford.py (hunk 1 above)
Comment (cclauss) [reply graphs/bellman_ford.py:1]: We want to use `dict, list` not `Dict, List`

Comment (mhihasan) [reply graphs/bellman_ford.py:1]: I haven't found this requirement at the issue #4052. Is it mentioned elsewhere?

File: graphs/bellman_ford.py (hunk 2 above)
Comment (cclauss) [reply graphs/bellman_ford.py:48]: Users trying again and again is OK.  This is how we all deal with Python on a regular basis.

File: graphs/bellman_ford.py (hunk 1 above)
Comment (cclauss) [reply graphs/bellman_ford.py:1]: #2464

Comment (mhihasan) [reply graphs/bellman_ford.py:1]: Ah, ok. Sorry , As I am new here, I couldn't notice that. I will keep that in mind for next PRs.

File: graphs/bellman_ford.py (hunk 3 above)
Comment (mhihasan) [reply graphs/bellman_ford.py:26]: Now, `edges` is being used only here. So, it has not been declared globally.

**Comment by mhihasan** [comment]
Are any other change  suggestions left?

**Review by cclauss — APPROVED** [review]

**Pull request merged**